src/main/java/pwr/asi/java/hermes/services/MailService.java

```diff
@@ -0,0 +1,62 @@
+package pwr.asi.java.hermes.services;
+
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import pwr.asi.java.hermes.dtos.MailDTO;
+import pwr.asi.java.hermes.entities.Mail;
+import pwr.asi.java.hermes.exceptions.InvalidDateException;
+import pwr.asi.java.hermes.repositories.MailRepository;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class MailService {
+
+    private final MailRepository mailRepository;
+
+    public MailService(MailRepository mailRepository) {
+        this.mailRepository = mailRepository;
+    }
+
+    private boolean resetCounter(Mail mailToCheck) {
+        final var lastMail = mailRepository.findFirst1ByOrderByCreationTimeDesc();
+        if (lastMail.isEmpty()) return false;
+        if (mailToCheck.getCreationTime().getYear() > lastMail.get().getCreationTime().getYear()) {
```

Comment:
You can return the result of this check, don't have to explicitly return true/false